File: Chapitre_2/exo2_5.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    int seconde,status1,status2;
    pid_t fils_1,fils_2;
    fils_1=fork();
    srand(time(NULL)+getpid());
    if(fils_1==0 ) {
        seconde=1 + (rand() % 10);
        sleep(seconde);
        exit(seconde);
    }
    fils_2=fork();
    if(fils_2==0) {
        seconde=1 + (rand() % 10);
        sleep(seconde);
        exit(seconde);
    }
    else if(fils_1!=0 && fils_2!=0) {
        wait(&status1);
        wait(&status2);
        printf("Duree totale :  %ds\n",WEXITSTATUS(status1)+WEXITSTATUS(status2));
    }
    if(fils_1==-1 || fils_2==-1) exit(1);
    return(0);
}

File: Chapitre_2/exo2_5.py
import os
import random
import time

fils_1=os.fork()
if fils_1==0:
    seconde=random.randint(1,10)
    time.sleep(seconde)
    os._exit(seconde)
fils_2=os.fork()
if fils_2==0:
    seconde=random.randint(1,10)
    time.sleep(seconde)
    os._exit(seconde)
if fils_1 !=0 and fils_2!=0:
    pid , status1=os.wait()
    pid , status2=os.wait()
    print("Durée totale : {}".format(os.WEXITSTATUS(status1)+os.WEXITSTATUS(status2)))
if fils_1 == -1 or fils_2 == -1:
    os._exit(1)